packages/@tinacms/fields/src/Wysiwyg/state/plugins/links/index.tsx

```diff
@@ -16,34 +16,55 @@ limitations under the License.
 
 */
 
-import { Fragment, Node, Schema, Slice } from 'prosemirror-model'
-import { Plugin, Transaction } from 'prosemirror-state'
-import { LinkFormController } from './LinkFormController'
-import { LinkView } from './LinkView'
+import { Fragment, Node, Slice, Mark } from 'prosemirror-model'
+import { Plugin, Transaction, PluginKey, EditorState } from 'prosemirror-state'
+import { EditorView } from 'prosemirror-view'
 
 export const HTTP_LINK_REGEX = /\bhttps?:\/\/(www\.)?[-a-zA-Z0-9@:%._\+~#=]{2,256}\.[a-z]{2,6}\b([-a-zA-Z0-9@:;%_\+.,~#?&//=]*)/g
 
-export function links(schema: Schema, theme?: any): Plugin {
-  const renderTarget = createInvisibleDiv('links')
-  let linkForm: LinkFormController
+interface LinkPluginState {
+  showLinkForm: boolean
+}
+
+export const linkPluginKey = new PluginKey('image')
+
+export function links(): Plugin {
   let shiftKey: boolean
   return new Plugin({
-    filterTransaction(tr: Transaction): boolean {
-      switch (tr.getMeta('type')) {
-        case 'tinacms/render':
-          linkForm.render(tr.getMeta('clickTarget'))
-          return false
-        case 'tinacms/unmount':
-          linkForm.unmount()
-          return false
-        default:
-          return true
-      }
-    },
-    view(editorView: any) {
-      insertElBefore(renderTarget, editorView.dom)
-      linkForm = new LinkFormController(renderTarget, editorView as any, theme)
-      return new LinkView(editorView as any, schema, renderTarget)
```

Comment:
State if link form is visible is saved in link plugin in this field `show_link_toolbar`.